### PR TITLE
refactor: workflow resilience polish and test observability

### LINKED ISSUE #114: refactor: workflow resilience polish and test observability
## Summary

CodeRabbit review on PR #111 identified several resilience and test quality improvements for the drift remediation and emergency change workflows. These are low-priority enhancements deferred to avoid scope creep.

### Resilience
- [ ] Add retry policy to `store_backup` activity calls (currently no retry — transient storage failures cause workflow failure without persisting backup)
- [ ] Guard post-revert `update_device_status` and `log_audit_event` calls in TTL success path — if status update fails after successful config revert, device has correct config but stale status
- [ ] Guard final `update_device_status` in emergency permanent path — same edge case as above

### Test observability
- [ ] Add spy/fixture tracking to verify `update_device_status` and `log_audit_event` are called with expected arguments in failure paths (currently tests only verify terminal exception message)
- [ ] Verify `rollback_config` execution in emergency deploy failure test (not just exception text)

## Context

- PR: #111
- All items are nitpick/trivial-level CodeRabbit suggestions, not bugs
- Current tests verify correct terminal behavior; these enhancements add defense-in-depth

### PR DESCRIPTION
**Pull request by anton-tvrz**
## Summary

Addresses #114 — low-priority CodeRabbit follow-ups from #111.

- **Resilience.** `store_backup` activity calls in both `DriftRemediationWorkflow` and `EmergencyChangeWorkflow` now use `device_retry_policy` so transient storage failures don't fail the workflow without persisting the backup. The two post-success housekeeping calls in `EmergencyChangeWorkflow` (final `update_device_status('active')` in the permanent path, and `update_device_status('active')` + `log_audit_event('EMERGENCY_REVERTED')` in the TTL-revert path) are now wrapped in try/except with a bounded retry policy — after the config change has already succeeded, a status-backend hiccup is logged but no longer fails the workflow.
- **Test observability.** Mock activities became spies that record their args; an autouse fixture clears them per test. Failure-path tests now assert that `update_device_status('maintenance')` and the matching `*_FAILED` audit event were actually called, and that `rollback_config` actually ran on emergency deploy/validation failure (previously tests only checked the terminal exception text). New tests added for the retry policy and the two new resilience guards.

## Test plan

- [x] `uv run invoke backend.test-unit` — 108/108 passing
- [x] `uv run invoke lint` and `uv run invoke format` — clean
- [x] Workflow file tests in isolation: 15/15 passing in 27s (was 335s on the Red run pre-fix — bounded retry policy is the speedup)
- [x] TDD cycle followed: failing tests committed first locally, then implementation made them green

Closes #114

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Added comprehensive test coverage for workflow failure scenarios and recovery paths.
  * Verified retry behavior for backup operations and transient failures.
  * Enhanced validation of audit event logging and device status tracking.

* **Bug Fixes**
  * Implemented retry policies for backup persistence operations.
  * Improved error handling in critical workflow paths to prevent cascading failures.

[Review Change Stack](https://app.coderabbit.ai/change-stack/anton-tvrz/project-network-synapse-quattro/pull/116?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: tests/unit/test_drift_remediation_workflow.py [hunk 1]
```diff
@@ -85,9 +85,23 @@ async def mock_validate_interfaces(device_hostname: str, ip_address: str, intend
...
+_recorded_audit_events: list[tuple[str, str, str]] = []
+_recorded_store_backup_calls: list[tuple[str, str]] = []
+
+
+@pytest.fixture(autouse=True)
+def _reset_spies() -> None:
+    _recorded_status_updates.clear()
+    _recorded_audit_events.clear()
+    _recorded_store_backup_calls.clear()
+
```
[the remark this answers is not in the document]
Comment (anton-tvrz) [reply tests/unit/test_drift_remediation_workflow.py]: Done in 0cfef31. Moved `_recorded_status_updates`, `_recorded_audit_events`, `_recorded_store_backup_calls`, and `_recorded_rollback_calls` plus the autouse reset fixture (now `_reset_workflow_spies`) into `tests/conftest.py`. Both workflow test modules import the lists from `tests.conftest` and dropped their local declarations.
